src/task-definitions.js

```diff
@@ -0,0 +1,347 @@
+import NotebookTask from './notebook-task'
+import { store } from './store'
+import actions from './actions'
+import { isCommandMode,
+  viewModeIsEditor,
+  getCellBelowSelectedId,
+  getCellAboveSelectedId } from './notebook-utils'
+
+const dispatcher = {}
+for (const action in actions) {
+  if (Object.prototype.hasOwnProperty.call(actions, action)) {
+    dispatcher[action] = (...params) => (store.dispatch(actions[action](...params)))
+  }
+}
+
+const oscpu = window.navigator.oscpu || window.navigator.platform
+let OSName = 'Unknown OS'
+if (oscpu.indexOf('Win') !== -1) OSName = 'Windows'
+if (oscpu.indexOf('Mac') !== -1) OSName = 'MacOS'
+if (oscpu.indexOf('X11') !== -1) OSName = 'UNIX'
+if (oscpu.indexOf('Linux') !== -1) OSName = 'Linux'
+
+function commandKey(key) {
+  let ctr = 'Ctrl '
+  if (OSName === 'MacOS') {
+    ctr = '⌘ '
+  }
+  return ctr + key
+}
+
+const TASKS = {}
+
+TASKS.EvaluateCell = new NotebookTask({
+  title: 'Evaluate Cell',
+  keybindings: ['mod+enter'],
+
+  keybindingCallback() {
+    dispatcher.changeMode('command')
+    dispatcher.evaluateCell()
+  },
+})
+
+TASKS.EvaluateAndSelectBelow = new NotebookTask({
+  title: 'Evaluate and Select Below',
+  keybindings: ['shift+enter'],
+
+  keybindingCallback() {
+    if (viewModeIsEditor()) {
+      dispatcher.changeMode('command')
+      dispatcher.evaluateCell()
+      const cellBelowId = getCellBelowSelectedId()
+      if (cellBelowId !== null) {
+        dispatcher.selectCell(cellBelowId, true)
+      } else {
+      // if cellBelowId *is* null, need to add a new cell.
+        dispatcher.addCell('javascript')
+        dispatcher.selectCell(getCellBelowSelectedId(), true)
+      }
+    }
+  },
+})
+
+TASKS.MoveCellUp = new NotebookTask({
+  title: 'Move Cell Up',
+  displayKeybinding: '\u21E7 \u2191',
+  keybindings: ['shift+up'],
+
+  callback() {
+    this.props.actions.cellUp()
+  },
+  keybindingCallback(e) {
+    if (isCommandMode()) {
+      if (e.preventDefault) {
+        e.preventDefault()
+      }
+      dispatcher.cellUp()
+    }
+  },
+})
+
+
+TASKS.MoveCellDown = new NotebookTask({
+  title: 'Move Cell Down',
+  displayKeybinding: '\u21E7 \u2193',
+  keybindings: ['shift+down'],
+
+  callback() {
+    this.props.actions.changeCellType('javascript')
+  },
+  keybindingCallback(e) {
+    if (isCommandMode()) {
+      if (e.preventDefault) {
+        e.preventDefault()
+      }
+      dispatcher.cellDown()
+    }
+  },
+})
+
+
+TASKS.SelectUp = new NotebookTask({
+  title: 'Select Cell Above',
+  displayKeybinding: '\u2191',
+  keybindings: ['up'],
+
+  keybindingCallback(e) {
+    if (isCommandMode()) {
+      // e.preventDefault blocks kbd scrolling of entire window
+      if (e.preventDefault) {
+        e.preventDefault()
+      } else { // internet explorer
+        e.returnValue = false
+      }
```

Comment:
this pattern:
```
      if (e.preventDefault) {
        e.preventDefault()
      } 
```
and this pattern:
```
   if (e.preventDefault) {
+        e.preventDefault()
+      } else { // internet explorer
+        e.returnValue = false
+      }
```

are showing up a lot. I should have cleaned it up when i last worked on keybindings-jupyter, but i think that (A) the latter can always safely replace the former (since apparently its an IE workaround) and (B) we can wrap the latter in a little function like:
```
function preventDefaultWrapper(e) {
    if (e.preventDefault) {
      e.preventDefault()
    } else {
      e.returnValue = false
   }
}
```

and just replace those 3 or 5 lines of cruft with a one liner `preventDefaultWrapper(e)` throughout the file. (If you want to go with this, you can of course call the function whatever you want

Reply:
YES. My comment above re: two callbacks is relevant here.